src/units/abilities.cpp

```diff
@@ -1547,31 +1554,31 @@ namespace { // Helpers for attack_type::special_active()
 //beneficiary unit does not have a corresponding weapon
 //(defense against ranged weapons abilities for a unit that only has melee attacks)
 
-unit_ability_list attack_type::get_weapon_ability(const std::string& ability) const
+unit_ability_list attack_type::get_weapon_ability(const std::string& ability, bool inverse_affect) const
 {
 	const map_location loc = self_ ? self_->get_location() : self_loc_;
 	unit_ability_list abil_list(loc);
 	if(self_) {
 		abil_list.append_if((*self_).get_abilities(ability, self_loc_), [&](const unit_ability& i) {
-			return special_active(*i.ability_cfg, AFFECT_SELF, ability, true);
+			return special_active(*i.ability_cfg, !inverse_affect ? AFFECT_SELF : AFFECT_OTHER, ability, true);
```

Comment:
`!inverse_affect ? AFFECT_SELF : AFFECT_OTHER` should be `inverse_affect ? AFFECT_OTHER : AFFECT_SELF`